### PR TITLE
Added Organizations dummy implementation

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/1504

### CODE REVIEW

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala [hunk 1]
```diff
@@ -105,6 +107,14 @@ trait Organizations {
       params: OrganizationSearchParams = OrganizationSearchParams.none
   ): UIO[UnscoredSearchResults[OrganizationResource]]
 
+  /**
+    * A non terminating stream of events for organizations. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def events(offset: Offset = NoOffset): Stream[Task, Envelope[OrganizationEvent]]
```
Comment (bogdanromanx) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala:116]: should we also add `currentEvents()` like in permissions?

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/realms/RealmsImpl.scala [hunk 2]
```diff
@@ -82,29 +81,17 @@ final class RealmsImpl private (
...
-          params.rev.forall(_ == rev) &&
-          params.types.subsetOf(types) &&
-          params.deprecated.forall {
-            _ == deprecated
-          }
-    }
-
-  def events(offset: Offset = NoOffset): Stream[Task, Envelope[RealmEvent]] =
+  override def events(offset: Offset = NoOffset): Stream[Task, Envelope[RealmEvent]] =
     eventLog.currentEventsByTag(entityType, offset)
```
Comment (bogdanromanx) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/realms/RealmsImpl.scala:93]: this should be `eventLog.eventsByTag`; the current func terminates

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        case (k, v) =>
+          k -> v.sortBy(_.sequenceNr)
+      }
+    }
+
+  override def create(
+      label: Label,
+      description: Option[String]
+  )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
+    eval(CreateOrganization(label, UUID.randomUUID(), description, caller)).flatMap(setUuid)
```
Comment (imsdu) [inline delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala:57]: In projects as generation uuid is a side effect just as getting current date, I removed the uuid from the create command and added an implicit `UIO[UUID]` besides the Clock

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala (hunk 1 above)
Comment (imsdu) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala:116]: I haven't added it in Realms either

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala (hunk 1 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Organizations.scala:116]: Not yet.

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/marshalling/QueryParamsUnmarshalling.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.delta.routes.marshalling
+
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.BaseUri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+
+trait QueryParamsUnmarshalling {
```
Comment (imsdu) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/marshalling/QueryParamsUnmarshalling.scala:8]: docs

**Review by bogdanromanx — APPROVED** [review]

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  * @param semaphore   a semaphore for serializing write operations on the journal
+  * @param maxStreamSize    truncate event stream after this size
+  */
+final class OrganizationsDummy private (
+    journal: IORef[OrganizationsJournal],
+    cache: IORef[OrganizationsCache],
+    uuidToLabel: IORef[UUIDToLabel],
+    semaphore: IOSemaphore,
+    maxStreamSize: Long
+)(implicit clock: Clock[UIO], uuidGenerator: UIO[UUID])
```
Comment (bogdanromanx) [inline delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala:40]: I think we should replace the `uuidGenerator: UIO[UUID]` to `uuidGenerator: () => UIO[UUID]`

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/config/ConfigReaderInstances.scala [hunk 6]
```diff
@@ -77,6 +80,18 @@ trait ConfigReaderInstances {
     }
   }
 
+  implicit final val indexingConfigReader: ConfigReader[IndexingConfig] = {
+    val logger: Logger = Logger[IndexingConfig]
+
+    @nowarn("cat=unused")
+    implicit val retryStrategyConfig: ConfigReader[RetryStrategy] =
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/config/ConfigReaderInstances.scala:87]: I think the `IndexingConfig` should only refer to `RetryStrategyConfig`, not an actual retry strategy. The retry strategy only makes sense within that module, based on a config.

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

**Review by imsdu — APPROVED** [review]

**Pull request merged**